### PR TITLE
GNSS time scales

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: astronomy/time_scales_body.hpp [hunk 1]
```diff
@@ -63,6 +63,14 @@ constexpr Instant FromTAI(quantities::Time const& tai) {
   return FromTT(tai + 32.184 * Second);
 }
 
+constexpr Instant FromGPSTime(quantities::Time const& gps) {
+  return FromTAI(gps + 19 * Second);
+}
+
+constexpr Instant From北斗Time(quantities::Time const& bdt) {
```
Comment (pleroy) [inline astronomy/time_scales_body.hpp]: Why `bdt` and not `北斗`?

File: astronomy/time_scales_body.hpp [hunk 2]
```diff
@@ -514,6 +532,14 @@ constexpr Instant operator""_UT1(char const* str, std::size_t size) {
   }
 }
 
+constexpr Instant operator""_GPS(char const* str, std::size_t size) {
```
Comment (pleroy) [inline astronomy/time_scales_body.hpp]: `const size`

File: astronomy/time_scales_body.hpp [hunk 3]
```diff
@@ -514,6 +532,14 @@ constexpr Instant operator""_UT1(char const* str, std::size_t size) {
...
+  return DateTimeAsGPSTime(operator""_DateTime(str, size));
+}
+
+constexpr Instant operator""_北斗(char const* str, std::size_t size) {
```
Comment (pleroy) [inline astronomy/time_scales_body.hpp]: Same.

File: astronomy/time_scales_test.cpp [hunk 4]
```diff
@@ -458,6 +458,23 @@ TEST_F(TimeScalesTest, EarthRotationAngle) {
...
 }
 
+TEST_F(TimeScalesTest, GNSS) {
+  // BeiDou Navigation Satellite System
+  // Signal In Space Interface Control Document
+  // Open Service Signals B1C and B2a (Test Version),
+  // 3.3 Time System.
+  // The start epoch of BDT is 00:00:00 on January 1, 2006 of Coordinated
+  // Universal Time (UTC).
+  EXPECT_THAT("2006-01-01T00:00:00"_北斗, Eq("2006-01-01T00:00:00"_UTC));
```
Comment (pleroy) [inline astronomy/time_scales_test.cpp]: Where are the 33 seconds?

Comment (eggrobin) [reply astronomy/time_scales_test.cpp]: Between UTC and TAI.

File: astronomy/time_scales_body.hpp (hunk 1 above)
Comment (eggrobin) [reply astronomy/time_scales_body.hpp]: Done.

File: astronomy/time_scales_body.hpp (hunk 2 above)
Comment (eggrobin) [reply astronomy/time_scales_body.hpp]: Done.

File: astronomy/time_scales_body.hpp (hunk 3 above)
Comment (eggrobin) [reply astronomy/time_scales_body.hpp]: Done.

**Pull request merged**